GameEngine/src/main/java/dk/sdu/se/f23/InVasion/gamestates/ShopState.java

```diff
@@ -72,16 +75,26 @@ public void draw() {
         int shopWidth= 200;
         sr.rect(1920-shopWidth,0,200,1080);
         sr.end();
-        sr.begin(ShapeRenderer.ShapeType.Filled);
-        sr.setColor(Color.GRAY);
+
+
+        SpriteBatch spriteBatch = new SpriteBatch();
```

Comment:
Vi har 1 spritebatch til hele projektet, brug den i stedet for at lave din egen, brug den istedet plzzz. Du kan få fat i den ved gameData.getSpriteBatch()

Reply:
Changed from SpriteBatch spriteBatch = new SpriteBatch();
to
SpriteBatch spriteBatch = gsm.getGameData().getSpriteBatch();

og nu virker det ikke længere.

kan det være at det er fordi den bliver lavet for sent og nu er null?

Reply:
I do believe so, just make a check that says `if gsm.getGameData().getSpriteBatch(); == null{
gsm.getGameData().setSpriteBatch(new SpriteBatch);}

else{
gsm.getGameData().getSpriteBatch();}`

Reply:
I do not know how, but this is both the most beautiful code and the most awful code at the same time XD